app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java

```diff
@@ -20,6 +22,11 @@ public boolean isSelected() {
         return selected;
     }
 
+    @Override
+    public boolean theSameAs(FilterContactTypeUi filterContactTypeUi) {
+        return false;
```

Comment:
Здесь нужно сравнить текущий объект с объектом из аргумента:
```suggestion
       return this.getContactType() == filterContactTypeUi.getContactType();
```